common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -318,20 +321,32 @@ def create(self, request, uuid):
                 }
             )
 
-        # Determine if this is a Switch session or a simple enroll and handle both.
         try:
-            course_run_string = CourseKey.from_string(course_run_id)
+            course_run_key = CourseKey.from_string(course_run_id)
         except InvalidKeyError:
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
                 data={
                     'message': 'Invalid {course_id}'.format(course_id=course_run_id)
                 }
             )
+
+        # Verify that the run is fullfillable
+        if not is_course_run_entitlement_fullfillable(course_run_key, datetime.datetime.now(UTC), entitlement):
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': 'The User is unable to enroll in Course Run {course_id} it is not available.'.format(
```

Comment:
Do we log these messages or just return them in the response? Could logging here be helpful? if so, should we include the username as well?

Reply:
The responses are logged.

Reply:
I do not think a log is required here. This is not a failure and more a failed attempt to enroll the user.  The 400 response will be logged though.